sw/device/silicon_creator/manuf/base/perso_tlv_data.c

```diff
@@ -63,29 +57,27 @@ status_t perso_tlv_set_cert_block(const uint8_t *buf, size_t max_room,
   if ((wrapped_cert_size < (name_len + min_header_size + 8)) ||
       (wrapped_cert_size > max_room))
     return INTERNAL();  // Something is really screwed up.
+  block->wrapped_cert_size = wrapped_cert_size;
+  block->wrapped_cert_p = buf;
 
-  memcpy(block->name, buf, name_len);
-  buf += name_len;
-  max_room -= name_len;
+  const uint8_t *name_start = cert_header_start + sizeof(crth);
+  memcpy(block->name, name_start, name_len);
   block->name[name_len] = '\0';
 
-  size_t cert_size;
-
-  cert_size = cert_x509_asn1_decode_size_header(buf);
-  if (cert_size > max_room)
-    return INTERNAL();
-
-  uint16_t wire_cert_size =
-      wrapped_cert_size - sizeof(perso_tlv_cert_header_t) - name_len;
-
-  if (cert_size != wire_cert_size) {
-    LOG_ERROR("Unexpected cert size %d instead of %d for cert %s", cert_size,
-              wire_cert_size, block->name);
-    return INTERNAL();
+  if (obj_type == kPersoObjectTypeX509Tbs ||
```

Comment:
I don't think you need to do the check twice. You can just do it when you hash the certs.

Reply:
Well I was thinking to validate the content wherever I change the parsing logic.
It's fine for me to remove the logic here.

Reply:
but when you hash the certs thats the final time a we are checking for correctness (after we have committed the certs to flash), so one check should be good enought right?